hash_tables/left_join.py

```diff
@@ -0,0 +1,25 @@
+"""Join two hash tables on the same key."""
+
+from .hash_table import HashTable
+
+
+def left_join(left_hash, right_hash):
+    """Join the right hash table to the left hash table on shared keys."""
+    joined = []
+
+    for bucket in left_hash.buckets:
+        # print(bucket)
+        if bucket.head:
```

Comment:
This only gets the first item out of the bucket. If there are multiple items in the bucket, you will not retrieve all the keys.